jwst/tso_photometry/tests/test_tso_photometry.py

```diff
@@ -0,0 +1,240 @@
+    radius_outer = None
+    data = mk_data_array(shape, value, background, xcenter, ycenter, radius)
+    datamodel = datamodels.CubeModel(data)
+    set_meta(datamodel, sub64p=True)
+
+    catalog = tso_aperture_photometry(datamodel, xcenter, ycenter,
+                                      radius, radius_inner,
+                                      radius_outer)
+
+    assert catalog.meta['instrument'] == datamodel.meta.instrument.name
+    assert catalog.meta['filter'] == datamodel.meta.instrument.filter
+    assert catalog.meta['subarray'] == datamodel.meta.subarray.name
+    assert catalog.meta['pupil'] == datamodel.meta.instrument.pupil
+    assert math.isclose(catalog.meta['xcenter'], xcenter, abs_tol=0.01)
+    assert math.isclose(catalog.meta['ycenter'], ycenter, abs_tol=0.01)
+
+    assert np.allclose(catalog['aperture_sum'],
+                       (value + background) * shape[-1] * shape[-2],
+                       rtol=1.e-6)
+    assert np.allclose(catalog['aperture_sum_err'], 0., atol=1.e-7)
+
+
+def test_tso_phot_3():
+
+    global shape, xcenter, ycenter
+
+    value = 17.
+    background = 0.8
+    radius = 5.
+    radius_inner = 8.
+    radius_outer = 11.
+    data = mk_data_array(shape, value, background, xcenter, ycenter, radius)
+    datamodel = datamodels.CubeModel(data)
+    set_meta(datamodel, sub64p=False)
+
+    int_times = include_int_times(datamodel)
+
+    catalog = tso_aperture_photometry(datamodel, xcenter, ycenter,
+                                      radius + 1., radius_inner,
+                                      radius_outer)
+
+    offset = datamodel.meta.exposure.integration_start - 1
+    slc = slice(offset, offset + shape[0])
+    assert np.allclose(catalog['MJD'], int_times[slc], atol=1.e-8)
+
+
+def test_tso_phot_4():
+
+    global shape, xcenter, ycenter
+
+    value = 17.
+    background = 0.8
+    radius = 5.
+    radius_inner = 8.
+    radius_outer = 11.
+    data = mk_data_array(shape, value, background, xcenter, ycenter, radius)
+    datamodel = datamodels.CubeModel(data)
+    set_meta(datamodel, sub64p=False)
+
+    int_times = include_int_times(datamodel)
+    # Modify the column of integration numbers so that they extend outside
+    # the range of integration numbers in the science data.  This shouldn't
+    # happen, i.e. this is to test an edge case.
+    int_start = datamodel.meta.exposure.integration_start
+    datamodel.int_times['integration_number'] += (2 * int_start)
+
+    catalog = tso_aperture_photometry(datamodel, xcenter, ycenter,
+                                      radius + 1., radius_inner,
+                                      radius_outer)
+
+    int_times = np.array([58704.63293, 58704.641845, 58704.65076,
+                          58704.65968, 58704.6686, 58704.677512,
+                          58704.686428])
+    assert np.allclose(catalog['MJD'], int_times, rtol=1.e-8)
+
+
+test_tso_phot_1()
+test_tso_phot_2()
+test_tso_phot_3()
+test_tso_phot_4()
```

Comment:
A test module shouldn't call the tests themselves on import.  Better to use `pytest test_tso_photometry.py` instead to do the test collection and running.

Reply:
Thanks @jdavies-st !  I added those for testing, and I forgot to delete them.  I'll do so now.